README.md

```diff
@@ -308,6 +313,238 @@ getPerson.addResolver(api, {
+- Target other AWS services like Lambda and other Event Buses
+
+Functionless uses a wrapped version of CDK's Event Bus, lets create a CDK event bus first.
+
+```ts
+// Create a new Event Bus using CDK.
+const awsBus = new aws_events.EventBus(this, "myBus");
+
+// Functionless also supports using the default bus or importing an Event Bus.
+const defaultBus = EventBus.fromEventBusName(this, "defaultBus", "default");
+const importedBus = EventBus.fromEventBusArn(this, "defaultBus", arn);
+```
+
+Functionless supports well typed events, lets add our event schema to Typescript.
+
+```ts
+interface UserDetails {
+  id?: string;
+  name: string;
+  age: number;
+  interests: string[];
+}
+
+interface UserEvent
+  extends functionless.EventBusRuleInput<
+    UserDetails,
+    // We can provide custom detail-types to match on
+    "Create" | "Update" | "Delete"
+  > {}
+```
+
+And wrap our CDK `EventBus`
+
+```ts
+const bus = functionless.EventBus<UserEvent>(awsEventBus);
+```
+
+#### Create Rules (`EventBusRule`) on a Event Bus to match incoming events.
+
+Now that you have a wrapped `EventBus`, lets add some rules.
+
+Functionless lets you write logic in Typescript on the type safe event.
+
+Lets match all of the create or updated events with one rule and another rule for deletes.
+
+```ts
+const createOrUpdateEvents = bus.when(
+  this,
+  "createOrUpdateRule",
+  (event) =>
+    event["detail-type"] === "Create" || event["detail-type"] === "Update"
+);
+const deleteEvents = bus.when(
+  this,
+  "deleteRule",
+  (event) => event["detail-type"] === "Delete"
+);
+```
+
+We also want to do something special when we get a new cat lover who is between 18 and 30 years old, lets make another rule for those.
+
+```ts
+const youngAdultCatLoversEvents = bus.when(
+  (event) =>
+    event["detail-type"] === "Create" &&
+    event.detail.interests.includes("CATS") &&
+    event.detail.age >= 18 &&
+    event.detail.age < 30
+);
+```
+
+#### Transform the event before sending to some services like `Lambda` Functions.
+
+We have two lambda functions to invoke, one for create or updates and another for deletes, lets make those.
+
+```ts
+const createOrUpdateFunction = new aws_lambda.Function(this, 'createOrUpdate', ...);
+const deleteFunction = new aws_lambda.Function(this, 'delete', ...);
+```
+
+and wrap them with Functionless's `Function` wrapper, including given them input types.
+
+```ts
+interface CreateOrUpdate {
+  id?: string;
+  name: string;
+  age: number;
+  operation: "Create" | "Update";
+  interests: string[];
+}
+
+interface Delete {
+  id: string;
+}
+
+const createOrUpdate = functionless.Function<CreateOrUpdate, void>(createOrUpdateFunction);
+const delete = functionless.Function<Delete, void>(createOrUpdateFunction);
+```
+
+The events from before do not match the formats from before, so lets transform them to the structures match.
+
+```ts
+const createOrUpdateEventsTransformed =
+  createOrUpdateEvents.map<CreateOrUpdate>((event) => ({
```

Comment:
I love the `.map<CreateOrUpdate>` - I usually try to avoid explicit type arguments but this solves the problem optimally.